### PR TITLE
Accessors in collection

### PR DESCRIPTION
**Pull request by Averethel**
```ruby
class Record < LHS::Record
  endpoint ':datastore/records`'
end
```

Response:
```json
{
  "items":  [{ "name": "Steve" }],
  "extra":  "extra",
  "total":  1
}
```

# Before
```ruby
Record.where.extra

# => 
# undefined method `extra' for #<LHS::Collection::InternalCollection::Collection:0x007f9b59b1a7c8>
```

# After
```ruby
Record.where.extra

# => 
# "extra"
```

### CODE REVIEW

File: lib/lhs/concerns/proxy/accessors.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+class LHS::Proxy
+  module Accessors
+    extend ActiveSupport::Concern
+
+    private
+
+    # Keywords that would not be forwarded via method missing
+    # FIXME: Extend the set of keywords
+    BLACKLISTED_KEYWORDS = %w(new proxy_association)
```
Comment (spape) [inline lib/lhs/concerns/proxy/accessors.rb]: Constants are only private if you use `private_constant`. I would put it on top. (minor)

File: lib/lhs/concerns/proxy/accessors.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      name = args.first if name == :[]
+      value = _data._raw[name.to_s]
+      if value.nil? && _data._raw.present?
+        value = _data._raw[name.to_sym]
+        value = _data._raw[name.to_s.classify.to_sym] if value.nil?
+      end
+      if value.is_a?(Hash)
+        handle_hash(value)
+      elsif value.is_a?(Array)
+        data = LHS::Data.new(value, _data)
```
Comment (spape) [inline lib/lhs/concerns/proxy/accessors.rb]: Suggestion: Refactor this block to `handle_collection` (symmetry) and rename the above method to `handle_item` to be correct *minor*

Comment (spape) [inline lib/lhs/concerns/proxy/accessors.rb]: and it the new context of accessor we might give a more precise name, like `return_item`, `return_collection` or `access_item` / `access_collection`

File: spec/collection/accessors_spec.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+describe LHS::Collection do
+  let(:datastore) { 'http://local.ch/v2' }
+  let(:items) { [{name: 'Steve'}] }
+  let(:extra) { 'extra' }
+  let(:collection) { Record.where }
```
Comment (spape) [inline spec/collection/accessors_spec.rb]: soon. `fetch`

File: spec/collection/accessors_spec.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    let(:response_data) do
+      {
+        items: items,
+        extra: extra,
+        total: 1
+      }
+    end
+
+    before(:each) do
+      LHC.config.placeholder('datastore', datastore)
```
Comment (spape) [inline spec/collection/accessors_spec.rb]: Try to not copy this outdated pattern. I know there are still a lot of tests with the additional placeholder configuration. But it's just an overhead I started some time ago to continue writing tests the following way:

```ruby
before(:each) do
class Record < LHS::Record
  endpoint 'http://datastore/records'
end
stub_request(:get, %r{http://datastore/v2/records})
  .to_return(body: response_data.to_json)
end
```

File: spec/collection/accessors_spec.rb (hunk 3 above)
Comment (spape) [inline spec/collection/accessors_spec.rb]: and BTW, this gives you the chain, not the collection ;)

**Comment by spape** [comment]
Rubocop ;P

**Comment by spape** [comment]
I recommend merging this before this major change: https://github.com/local-ch/lhs/pull/190

File: spec/collection/accessors_spec.rb (hunk 4 above)
Comment (Averethel) [inline spec/collection/accessors_spec.rb]: yeah I wasn't sure which one is the current state of art so I used some other collection spec as a template.

File: spec/collection/accessors_spec.rb (hunk 3 above)
Comment (Averethel) [inline spec/collection/accessors_spec.rb]: True but it gets resolved on access no?

**Comment by Averethel** [comment]
@spape rubocop? what ruboccop? :trollface:

**Pull request merged**